src/cls/rbd/cls_rbd.cc

```diff
@@ -3087,7 +3087,13 @@ int image_set(cls_method_context_t hctx, const string &image_id,
     std::string global_id_key = global_key(mirror_image.global_image_id);
     std::string image_id;
     r = read_key(hctx, global_id_key, &image_id);
-    if (r != -ENOENT) {
+    if (r < 0) {
```

Comment:
Instead of the nested 'if' statements, how about something like:

```
if (r >= 0) {
  return -EEXIST;
} else if (r != -ENOENT) {
  CLR_ERR(...)
  return r;
}
```